src/neuroglancer/save_state/save_state.ts

```diff
@@ -0,0 +1,333 @@
+import 'neuroglancer/save_state/save_state.css';
+
+import {debounce} from 'lodash';
+
+import {Overlay} from 'neuroglancer/overlay';
+import {getOldStyleSaving} from 'neuroglancer/preferences/user_preferences';
+import {StatusMessage} from 'neuroglancer/status';
+import {RefCounted} from 'neuroglancer/util/disposable';
+import {cancellableFetchOk, responseJson} from 'neuroglancer/util/http_request';
+import {getRandomHexString} from 'neuroglancer/util/random';
+import {Trackable} from 'neuroglancer/util/trackable';
+import {Viewer} from 'neuroglancer/viewer';
+
+export class SaveState extends RefCounted {
+  private activeKey?: string|null;
+  lastKey?: string|null;
+  saveStorage: any = {};
+  supported = true;
+  constructor(public root: Trackable, updateDelayMilliseconds = 400) {
+    super();
+    const saverManuallyDisabled = !getOldStyleSaving().value;
+    this.supported = !storageAvailable() && saverManuallyDisabled;
```

Comment:
This seems confusing - state saving is supported when there is no storage available and the user manually disabled state saving? Either the check or the variable names don't match...